### PR TITLE
Update dependencies to latest versions

### PR DESCRIPTION
**Pull request by mraible**
Updated with `npx npm-check-updates`. 

```
 @crowdstrike/foundry-js                    0.9.0  →    0.15.0
 @shoelace-style/shoelace                   2.5.2  →    2.16.0
 @testing-library/jest-dom                 ^6.1.3  →    ^6.5.0
 @testing-library/react                   ^14.0.0  →   ^16.0.1
 @types/react                            ^18.2.14  →   ^18.3.5
 @types/react-dom                         ^18.2.6  →   ^18.3.0
 @typescript-eslint/eslint-plugin         ^5.61.0  →    ^8.5.0
 @typescript-eslint/parser                ^5.61.0  →    ^8.5.0
 @vitejs/plugin-react                      ^4.0.1  →    ^4.3.1
 autoprefixer                            ^10.4.14  →  ^10.4.20
 eslint                                   ^8.44.0  →   ^9.10.0
 eslint-plugin-no-relative-import-paths    ^1.5.2  →    ^1.5.5
 eslint-plugin-prettier                    ^5.0.0  →    ^5.2.1
 eslint-plugin-react                      ^7.33.1  →   ^7.36.1
 eslint-plugin-react-hooks                 ^4.6.0  →    ^4.6.2
 eslint-plugin-react-refresh               ^0.4.1  →   ^0.4.11
 framer-motion                           ^10.13.1  →   ^11.5.4
 jsdom                                    ^22.1.0  →   ^25.0.0
 postcss                                  ^8.4.26  →   ^8.4.45
 prettier                                   3.0.0  →     3.3.3
 react                                    ^18.2.0  →   ^18.3.1
 react-dom                                ^18.2.0  →   ^18.3.1
 react-hook-form                          ^7.45.2  →   ^7.53.0
 react-router-dom                          6.14.1  →    6.26.2
 react-select                              ^5.7.4  →    ^5.8.0
 rollup-plugin-visualizer                  ^5.9.2  →   ^5.12.0
 tailwindcss                               ^3.3.3  →   ^3.4.11
 temporal-polyfill                         ^0.1.1  →    ^0.2.5
 typescript                                ^5.0.2  →    ^5.6.2
 vite                                      ^4.4.0  →    ^5.4.4
 vitest                                   ^0.34.5  →    ^2.1.0
 zod                                      ^3.21.4  →   ^3.23.8
```

### CODE REVIEW

File: ui/pages/scalable-rtr-react/.yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+nodeLinker: node-modules
```
Comment (mraible) [inline ui/pages/scalable-rtr-react/.yarnrc.yml:1]: Not sure if the two files above should be checked in. If not, we should ignore them.

**Comment by mraible** [comment]
`yarn install` seems to work after upgrading these libraries, but there are warnings:

```
➤ YN0000: · Yarn 4.3.1
➤ YN0000: ┌ Resolution step
➤ YN0085: │ + @crowdstrike/foundry-js@npm:0.15.0, @shoelace-style/shoelace@npm:2.16.0, and 242 more.
➤ YN0085: │ - @adobe/css-tools@npm:4.3.1, @babel/compat-data@npm:7.23.2, @babel/core@npm:7.23.2, and 236 more.
➤ YN0000: └ Completed in 4s 999ms
➤ YN0000: ┌ Post-resolution validation
➤ YN0060: │ eslint is listed by your project with version 9.10.0 (pbabda), which doesn't satisfy what @typescript-eslint/eslint-plugin and other dependencies request (^8.57.0).
➤ YN0002: │ scalable-rtr-react@workspace:. doesn't provide @testing-library/dom (p02ebd), requested by @testing-library/react.
➤ YN0086: │ Some peer dependencies are incorrectly met by your project; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code.
➤ YN0086: │ Some peer dependencies are incorrectly met by dependencies; run yarn explain peer-requirements for details.
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0066: │ typescript@patch:typescript@npm%3A5.6.2#optional!builtin<compat/typescript>::version=5.6.2&hash=379a07: Cannot apply hunk #11
➤ YN0013: │ 113 packages were added to the project (+ 113.45 MiB).
➤ YN0000: └ Completed in 3s 257ms
➤ YN0000: ┌ Link step
➤ YN0007: │ esbuild@npm:0.21.5 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 5s 181ms
➤ YN0000: · Done with warnings in 13s 482ms
```

If I try to run `yarn test`, it fails:

```
 DEV  v2.1.0 /Users/mraible/dev/foundry-sample-scalable-rtr/ui/pages/scalable-rtr-react

 ❯ src/components/ZonedDateTime.test.tsx (0)
 ❯ src/lib/utils/cron-parser.test.ts (0)
 ❯ src/lib/utils/datetime.test.ts (0)
 ❯ src/lib/utils/truncate-text.test.ts (0)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Suites 4 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/components/ZonedDateTime.test.tsx [ src/components/ZonedDateTime.test.tsx ]
 FAIL  src/lib/utils/cron-parser.test.ts [ src/lib/utils/cron-parser.test.ts ]
 FAIL  src/lib/utils/datetime.test.ts [ src/lib/utils/datetime.test.ts ]
 FAIL  src/lib/utils/truncate-text.test.ts [ src/lib/utils/truncate-text.test.ts ]
Error: Cannot find module '@testing-library/dom'
Require stack:
- /Users/mraible/dev/foundry-sample-scalable-rtr/ui/pages/scalable-rtr-react/node_modules/@testing-library/react/dist/pure.js
 ❯ Object.<anonymous> node_modules/@testing-library/react/dist/pure.js:46:12

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/4]⎯

 Test Files  4 failed (4)
      Tests  no tests
   Start at  11:51:10
   Duration  595ms (transform 17ms, setup 0ms, collect 0ms, tests 0ms, environment 1.22s, prepare 474ms)
```

To fix this, I ran `yarn add -D @testing-library/dom`. Now some three tests fail:

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 3 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/lib/utils/datetime.test.ts > toZoned method > Bad inputs > Malformed timestamp
AssertionError: expected error to have message: Cannot parse dateTime 'asdf[UTC]'

Expected: "Cannot parse dateTime 'asdf[UTC]'"
Received: "Cannot parse: asdf[UTC]"

 ❯ src/lib/utils/datetime.test.ts:61:10
     59|       expect(() => {
     60|         toZoned({ timestamp: "asdf" });
     61|       }).toThrow(new RangeError("Cannot parse dateTime 'asdf[UTC]'"));
       |          ^
     62|     });
     63|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/3]⎯

 FAIL  src/lib/utils/datetime.test.ts > toZoned method > Bad inputs > Invalid date
AssertionError: expected error to have message: Invalid overflowed value 13

Expected: "Invalid overflowed value 13"
Received: "Invalid isoMonth: 13; must be between 1-12"

 ❯ src/lib/utils/datetime.test.ts:67:10
     65|       expect(() => {
     66|         toZoned({ timestamp: "2023-13-32T25:61" });
     67|       }).toThrow(new RangeError("Invalid overflowed value 13"));
       |          ^
     68|     });
     69|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/3]⎯

 FAIL  src/lib/utils/datetime.test.ts > toZoned method > Bad inputs > Invalid timezone
AssertionError: expected error to have message: Invalid time zone specified: Moon/Sea_Of_Tranquility

Expected: "Invalid time zone specified: Moon/Sea_Of_Tranquility"
Received: "Invalid time zone specified: MOON/SEA_OF_TRANQUILITY"

 ❯ src/lib/utils/datetime.test.ts:73:10
     71|       expect(() => {
     72|         toZoned({ timestamp: summerDate, timezone: "Moon/Sea_Of_Tranquility" });
     73|       }).toThrow(
       |          ^
     74|         new RangeError("Invalid time zone specified: Moon/Sea_Of_Tranquility"),
     75|       );

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/3]⎯

 Test Files  1 failed | 3 passed (4)
      Tests  3 failed | 37 passed (40)
```

File: ui/pages/scalable-rtr-react/.yarnrc.yml (hunk 1 above)
Comment (prvn) [reply ui/pages/scalable-rtr-react/.yarnrc.yml:1]: probably not, we should ignore.

**Review by prvn — APPROVED** [review]
LGTM, add `install-state.gz` and `.yarnrc.yml` to gitignore and you can merge this

**Pull request merged**